config/software/ruby.rb

```diff
@@ -170,51 +171,29 @@
   # Also, fix paths emitted in the makefile on windows on both msys and msys2.
   patch source: "ruby-mkmf.patch", plevel: 1, env: patch_env
 
-  # RHEL 6's gcc doesn't support `#pragma GCC diagnostic` inside functions, so
-  # we'll guard their inclusion more specifically. As of 2018-01-25 this is fixed
-  # upstream and ought to be in 2.5.1
-  if rhel? &&
-      platform_version.satisfies?("< 7") &&
-      (version == "2.5.0")
-    patch source: "prelude_25_el6_no_pragma.patch", plevel: 0, env: patch_env
-  end
-
-  # Backporting a 2.6.0 fix to 2.5.1 (and 2.4.4 for ChefDK 2). This allows us to build Nokogiri 1.8.3.
-  # Basically we only include `-Werror` linker warnings when building native gems if we are on Windows.
-  # This prevents some "expected" warnings from failing the build.
-  if version == "2.5.1" || version == "2.4.4"
-    patch source: "ruby-only-compiler-warnings-on-windows.patch", plevel: 1, env: patch_env
-  end
-
   configure_command = ["--with-out-ext=dbm,readline",
                        "--enable-shared",
-                       "--enable-load-relative",
```

Comment:
@jmartin-r7 This was enabled intentionally with enabling portable builds: https://github.com/rapid7/metasploit-omnibus/pull/127

Commit:
https://github.com/rapid7/metasploit-omnibus/commit/4e4904663045a18537e1ddd149d18a780db64a3d#diff-8ff31cf08a88774679ed15595ec42f8fea9608671ca50be341953489045e8e25R186

Unless enabled-load-relative is enabled by default in Ruby 2.7.2, I think we'll have to keep this line in 👀 

I can't think of a cleaner way to ensure this doesn't get accidentally deleted again in the future - other than adding a quick test for the configuration value being set maybe?

Reply:
I will restore, I must have missed this when porting in the latest upstream version of the file.